turbopack/crates/turbopack-core/src/issue/mod.rs

```diff
@@ -512,6 +534,44 @@ impl IssueSource {
     }
 }
 
+async fn source_pos(
+    source: Vc<Box<dyn Source>>,
+    start: SourcePos,
+    end: SourcePos,
+) -> Result<Option<(Vc<Box<dyn Source>>, SourcePos, SourcePos)>> {
+    let Some(generator) = Vc::try_resolve_sidecast::<Box<dyn GenerateSourceMap>>(source).await?
+    else {
+        return Ok(None);
+    };
+
+    let Some(srcmap) = &*generator.generate_source_map().await? else {
+        return Ok(None);
+    };
+
+    let find = |line: usize, col: usize| async move {
+        let token = srcmap.lookup_token(line, col).await?;
+
+        match &*token {
+            crate::source_map::Token::Synthetic(t) => Ok::<_, anyhow::Error>(SourcePos {
+                line: t.generated_line as _,
+                column: t.generated_column as _,
+            }),
+            crate::source_map::Token::Original(t) => Ok(SourcePos {
+                line: t.original_line as _,
+                column: t.original_column as _,
+            }),
+        }
+    };
+
+    let start = find(start.line, start.column).await?;
+    let end = find(end.line, end.column).await?;
+
+    let new_source = *generator.original_source().await?;
```

Comment:
Instead of this method you need to lookup the source from the SourceMap.

A SourceMap could references multiple source files. It's not guaranteed that there is only one.

You also need to lookup the source content from the source map.

You might also need to call `convert_to_turbopack_source_map` on the source map to make sure the source map contains the correct source format and all sources has content.